### DISCUSSION TITLE
PNPMのエラーでMisskey 2024.2.0アップデート後サーバーが落ちている。「MK_DEV_PREFER=backend pnpm dev」が実行出来ない。

### DISCUSSION THREAD

**Discussion by oita-pikapika** [misskey-dev/misskey#13521] [Q&A]
プログラミング初心者の、最近サーバーを立ち上げたばかりの者です。Misskeyを2024.2.0にアップデートしたところからサーバーがダウンしててしまい、自力で調べて解決しようと試みましたがとう詰んでしまったので、質問させていただきにきました。
お手数おかけしますが、初心者にも分かるようにお教えいただけますとうれしいです。

【状況】
Misskeyをアップデートしようとすると、

Failed in 32.9s at /home/misskey/misskey/packages/frontend
(間が空いて)
ERR_PNPM_RESCURSIVE_RUN_FIRST_FAIL frontend@build:`vite build`
Exit status 134
ELIFECYLCLE Command failed with exit code 1.

と表示され、きちんとアップデート出来ず、サーバーが落ちる。

【自分で試したこと】
https://github.com/misskey-dev/misskey/issues/10957
を参考に、/home/misskey/misskey/packages/frontendのpackage.json にtype scriptのインストールをする
↓
解決出来ない。

https://github.com/misskey-dev/misskey/issues/12861
を参考に、/home/misskey/misskey/packages/frontend内にて「MK_DEV_PREFER=backend pnpm dev」を実行しようとする。
「Command "dev" not found」と出てきてしまい、実行できない。/home/misskey/misskey/packages/frontendのpackage.json内にて、devを定義しようとしたりするも、別のエラーが出てしまい、失敗。
「pnpm build-assets」も検索しても何か分からず、実行できない。

ここで自力で検索するのでは限界がきて、詰んでしまっています。
どなたかお力をお貸しいただけますと大変嬉しいです。どうかよろしくお願いいたします。

**Comment by tamaina** [comment]
- 何がしたいのかよくわからないのですが、/home/misskey/misskey/で`MK_DEV_PREFER=backend pnpm --filter frontend run dev`をしてみてください
  * MK_DEV_PREFERとfrontendのrun devは両立しないものなので無意味だと思うのですが…
  * そもそもfrontendのpnpm run dev/watchだけを起動したところでMisskeyは動かない

**Reply by tamaina** [reply]
開発環境を立ち上げるだけなら/home/misskey/misskey/で`pnpm dev`もしくは`MK_DEV_PREFER=backend pnpm dev`を実行するだけなのですが、なぜfrontendで実行しようとしたのですか？

**Reply by tamaina** [reply]
あっ違った、Failed in 32.9s at /home/misskey/misskey/packages/frontendはメッセージだわ…

**Comment by tamaina** [comment]
https://github.com/misskey-dev/misskey/issues/10957 では

> Resolved by installing global typescript compiler

とあるので、`sudo npm install typescript -g`でグローバルにtypescriptをインストールしてみてください。

**Reply by oita-pikapika** [reply]
ご返信ありがとうございます。`pnpm dev`と`MK_DEV_PREFER=backend pnpm dev`を/home/misskey/misskey/で実行しようとしたところ、「Command "dev" not found」のエラーが出てしまいました。
また、`sudo npm install typescript -g`を実行した後再び`update_misskey`してみたところ、「ERR_PNPM_RESCURSIVE_RUN_FIRST_FAIL frontend@build:vite build
Exit status 134
ELIFECYLCLE Command failed with exit code 1.」
と同じエラーが出てしまいました。
